modules/tensor_mechanics/include/materials/ADGeneralizedRadialReturnStressUpdate.h

```diff
@@ -0,0 +1,162 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "StressUpdateBase.h"
+#include "ADGeneralizedReturnMappingSolution.h"
+
+/**
+ * ADGeneralizedRadialReturnStressUpdate computes the generalized radial return stress increment for
+ * anisotropic (Hill-like) creep and plasticity.  This radial return mapping class
+ * acts as a base class for the radial return creep and plasticity classes/combinations.
+ * The stress increment computed by ADRadialReturnStressUpdate is used by
+ * ComputeMultipleInelasticStress which computes the elastic stress for finite
+ * strains.
+ * This class is based on the algorithm in Versino, D, Bennett, KC. Generalized
+ * radial return mapping algorithm for anisotropic von Mises plasticity framed in material
+ * eigenspace. Int J Numer Methods Eng. 2018. 116. 202 222
+ */
+
+#include "MooseTypes.h"
+
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
+#include <Eigen/Dense>
+#include "metaphysicl/dualnumber.h"
+
+namespace Eigen
```

Comment:
can we move this into the implementation file? or into MooseTypes.h

Reply:
Every child class would need to do this if we move it to the implementation file. Is this so bad?